src/rantanplan/structures.py

```diff
@@ -11,230 +10,155 @@
 # )
 # Structures will be checked in order of definition, the first one to match
 # will be chosen.
+ASSONANT_RHYME = "assonant"
+CONSONANT_RHYME = "consonant"
+ARTE_MAYOR_MIN_LENGTH = 9
+ARTE_MENOR_MAX_LENGTH = 8
+COPLA_ARTE_MENOR_MIN_LENGTH = 4
+LIRA_LONG_LINE = 11
+LIRA_SHORT_LINE = 7
+SILVA_LONG_LINE = LIRA_LONG_LINE
+SILVA_SHORT_LINE = LIRA_SHORT_LINE
+HENDECASYLLABLE = 11
+ALEXANDRINE = 14
+OCTOSYLLABLE = ARTE_MENOR_MAX_LENGTH
+TETRASYLLABLE = COPLA_ARTE_MENOR_MIN_LENGTH
+MINIMUM_SAFE_LENGTH = 3
+MAXIMUM_SAFE_LENGTH = 20
+#    STRUCTURES_LENGTH is a dictionary with a list of line lengths and a `True`
+#    or `False` flag to indicate if the structure can be cyclic or not
+STRUCTURES_LENGTH = {
+    "seguidilla_compuesta": ([7, 5, 7, 5, 5, 7, 5], False),
+    "seguidilla": ([7, 5, 7, 5], False),
+    "chamberga": ([7, 5, 7, 5, 3, 7, 3, 7, 3, 7], False),
+    "seguidilla_gitana": ([6, 6, 11, 6], False),
+    "estrofa_safica": ([11, 11, 11, 5], False),
+    "estrofa_unamuno": ([11, 11, 7, 5], False),
+    "estrofa_francisco_de_la_torre": ([11, 11, 11, 7], False),
+    "endecha_real": ([7, 7, 7, 11], True),
+    "lira": ([7, 11, 7, 7, 11], False),
+    "estrofa_manriqueña": ([8, 8, 4, 8, 8, 4], False),
+    "ovillejo_tetra": ([8, 4, 8, 4, 8, 4, 8, 8, 8, 8], False),
+    "ovillejo_tri": ([8, 3, 8, 3, 8, 3, 8, 8, 8, 8], False),
+    "haiku": ([5, 7, 5], False),
+    "sonnet": (14*[11], False),
+}
 
 
-def most_common_lengths(lengths_list, n=4, number_of_verses=None):
-    """Return the n most commons lengths in a list of verses lengths
+def has_minimum_length(min_length, ranges_list):
+    """Checks if every range within a list of ranges contains a minimum length
 
-    :param lengths_list: List with the lengths to be checked
-    :param n: Number of most common lengths to be returned
-    :param number_of_verses: How many verses are in a given stanza
-    :return: The n most common lengths
+    :param min_length: Minimum length
+    :param ranges_list: List of verse length ranges
+    :return:
     """
-    most_common_lengths_list = []
-    if len(lengths_list) == number_of_verses or number_of_verses is None:
-        count = Counter(lengths_list).most_common(n)
-        for length, _ in count:
-            # `lengths` is a tuple with the length and it's count so we only get
-            # the first element of the tuple
-            most_common_lengths_list.append(length)
-    return most_common_lengths_list
+    return all(
+        (min_length <= max(length_range) for length_range in ranges_list))
```

Comment:
I'm pretty sure I've reviewed this already.
```suggestion
    return all(min_length <= max(length_range) for length_range in ranges_list)
```